### PR TITLE
Add Transaction Submission Tutorial

### PR DESCRIPTION
**Pull request by delventhalz**
Adds a tutorial for building Transactions/Batches, including signing, with example code from Python, JavaScript, and the JavaScript SDK. Readers can switch between example code using a dropdown in the compiled html.

### CODE REVIEW

File: docs/source/app_developers_guide/txn_submit_tutorial/_segments/keys_public.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+In addition to a private key, you will need a public key encoded as a hexadecimal string to distribute with the Transaction and confirm its signature. This key must must be generated from the same private key you used in signing.
```
Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/keys_public.rst]: "need the public key associated with the private key", not just any public key.

File: docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Encoding your Payload
+=====================
+
+Transaction payloads are composed of some sort of raw binary that is opaque to the validator. The logic for decoding them rests entirely within the particular Transaction Processor itself. As a result, there are many possible formats, and you will have to look to the definition of the TP for that information. But as an example, the *IntKey* Transaction Processor uses a payload of three key/value pairs encoded as CBOR. Creating one might look like this:
```
Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst]: "some sort of raw binary" to "binary-encoded data"

Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst]: add "encoding and" before "decoding"

Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst]: Suggest linking CBOR to the wiki entry for it.

File: docs/source/app_developers_guide/txn_submit_tutorial/_segments/header_info.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Remember from :doc:`/architecture/transactions_and_batches`, that *inputs* and *outputs* are state addresses that the Transaction is allowed to read from or write to. In this case we used a six character prefix rather than full 70-character addresses, allowing us to work with any address in the IntKey subtree. Also remember that *dependencies* are the *header signatures* of Transactions that must be committed before ours (none in this case).
```
Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/header_info.rst]: Not sure I would phrase this reference this way.  It implies that the app developer guide reader is the same audience as the architecture doc reader.

File: docs/source/app_developers_guide/txn_submit_tutorial/_segments/batch5_encode.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+5. Encode the Batch(es)
+-----------------------
+
+In order to transmit one or more Batches to a validator for submission, they must be serialized, and *BatchList* is the protobuf message class for that purpose. It has one *batches* property, which should be set to one or more Batches.
```
Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/batch5_encode.rst]: change the end of the sentence to "serialized in a *BatchList* protobuf message."

**Comment by delventhalz** [comment]
The source code here is divided up to allow for example code in new languages to easily be plugged in. This unfortunately means it is not great to read before being built.

Here is a link to the built html:
https://build.sawtooth.me/job/Sawtooth-Hyperledger/job/sawtooth-core/job/PR-381/1/artifact/docs/build/html/app_developers_guide/txn_submit_tutorial/python.html

File: docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Encoding your Payload
+=====================
+
+Transaction payloads are composed of binary-encoded data that is opaque to the validator. The logic for encoding and decoding them rests entirely within the particular Transaction Processor itself. As a result, there are many possible formats, and you will have to look to the definition of the TP itself for that information. But as an example, the *IntKey* Transaction Processor uses a payload of three key/value pairs encoded as `CBOR <https://en.wikipedia.org/wiki/CBOR>`_. Creating one might look like this:
```
Comment (peterschwarz) [inline docs/source/app_developers_guide/txn_submit_tutorial/_segments/payload.rst]: Replace "TP" with "Transaction Processor"

**Pull request closed without merging**